src/main/java/org/reaktivity/nukleus/auth/jwt/internal/stream/ProxyStreamFactory.java

```diff
@@ -0,0 +1,400 @@
+        beginEx.headers().forEach(h ->
+        {
+            if (BufferUtil.equals(h.name(), AUTHORIZATION))
+            {
+                String16FW authorizationHeader = h.value();
+                final DirectBuffer buffer = authorizationHeader.buffer();
+                final int limit = authorizationHeader.limit();
+                int offset = BufferUtil.limitOfBytes(buffer, authorizationHeader.offset(),
+                        limit, BEARER_PREFIX);
+                if (offset > 0)
+                {
+                    String token = buffer.getStringWithoutLengthUtf8(offset, limit - offset);
+                    String realm = validator.validateAndGetRealm(token);
+                    if (realm != null)
+                    {
+                        authorization[0] = supplyRealmId.applyAsLong(realm);
+                    }
+                }
+            }
+        });
+         return authorization[0];
+    }
+
+    private MessageConsumer newConnectReplyStream(
+            final BeginFW begin,
+            final MessageConsumer throttle)
+    {
+        final long throttleId = begin.streamId();
+
+        return new ProxyConnectReplyStream(throttle, throttleId)::handleStream;
+    }
+
+    private RouteFW wrapRoute(
+            int msgTypeId,
+            DirectBuffer buffer,
+            int index,
+            int length)
+    {
+        return routeRO.wrap(buffer, index, index + length);
+    }
+
+    abstract class ProxyStream
+    {
+        private final MessageConsumer sourceThrottle;
+        private final long sourceStreamId;
+
+        MessageConsumer target;
+        long targetStreamId;
+
+        private MessageConsumer streamState;
+
+        private ProxyStream(
+                MessageConsumer sourceThrottle,
+                long sourceStreamId)
+        {
+            this.sourceThrottle = sourceThrottle;
+            this.sourceStreamId = sourceStreamId;
+            this.streamState = this::beforeBegin;
+        }
+
+        void handleStream(
+                int msgTypeId,
+                DirectBuffer buffer,
+                int index,
+                int length)
+        {
+            streamState.accept(msgTypeId, buffer, index, length);
+        }
+
+        private void beforeBegin(
+                int msgTypeId,
+                DirectBuffer buffer,
+                int index,
+                int length)
+        {
+            if (msgTypeId == BeginFW.TYPE_ID)
+            {
+                final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+                handleBegin(begin);
+            }
+            else
+            {
+                writer.doReset(sourceThrottle, sourceStreamId);
+            }
+        }
+
+        private void handleBegin(BeginFW begin)
+        {
+            doHandleBegin(begin);
+            this.streamState = this::afterBegin;
+        }
+
+        abstract void doHandleBegin(
+            BeginFW begin);
```

Comment:
Rather than inheritance with override for `doHandleBegin` why not manage the `BEGIN` in the `newStream(...)` method instead, handled differently for accept or connect-reply and then have a single implementation for `ProxyStream`?

Reply:
Done